### PR TITLE
feat: add linting (with changes)

### PR DESCRIPTION
**Pull request by JcdeA**
Messed up the branch for #13  so here's a new branch with the changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,17 @@
         "start": "concurrently --kill-others \"cd lib/destack && npm run dev\" \"cd dev/nextjs-project && npm run dev\""
```
Comment (LiveDuo) [inline package.json:9]: Add the `prepare` script? (Got it from #14 )
```js
"prepare": "husky install"
```

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx pretty-quick
```
Comment (LiveDuo) [inline .husky/pre-commit:4]: I've played around with `husky` a bit and it worked nicely so far. Will be good to add `--staged` to check staged files as well!

```sh
npx pretty-quick --staged
```

**Review by LiveDuo — CHANGES_REQUESTED** [review]
I think `.husky/.gitignore` is also missing but not sure if that's require for husky to work. If it's not we can merge when the `prepare` script and the `--staged` flag are added.

File: package.json (hunk 1 above)
Comment (JcdeA) [reply package.json:9]: Ahh, forgot about that

Comment (LiveDuo) [reply package.json:9]: No worries!

**Comment by JcdeA** [comment]
@LiveDuo How do I fix this conflict..?
The one I commit is the one we need because I added lint-staged as a devdependency.

**Comment by JcdeA** [comment]
Shooot looks like the latest commit was a mistake.. 

Typescript hasn't been merged yet, so tsc is not a dependency yet in this branch..

**Comment by LiveDuo** [comment]
Make push another commit to undo adding `tsc`. I think it should work!

Don't worry about `package-lock` I'll sort it out when the PR is merged.

**Comment by JcdeA** [comment]
Thanks!

**Comment by JcdeA** [comment]
I think this is good to merge now!

**Pull request merged**